src/hotspot/os/linux/systemMemoryBarrier_linux.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+#include "runtime/os.hpp"
+#include "logging/log.hpp"
+#include "utilities/debug.hpp"
+#include "utilities/systemMemoryBarrier.hpp"
+
+#if defined(AMD64) || defined(AARCH64)
+
+#include <sys/syscall.h>
+// Syscall defined in kernel 4.3
+#if !defined(SYS_membarrier)
+#define SYS_membarrier 324
```

Comment:
SYS_gettid just have large ifdef else and then an error.
I guess e.g. ppc always compile on platforms with it defined since it doesn't need to define it.

The solution with a per plattform file as zSyscall does seems to be the nicest way. But IMHO a bit out of scope for this PR.

Reply:
@TheRealMDoerr please have another look :)